src/charm.py

```diff
@@ -16,73 +18,40 @@
 )
 
 
-def gen_pass() -> str:
-    return "".join(choices(ascii_uppercase + digits, k=30))
-
-
 class Operator(CharmBase):
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
         self.log = logging.getLogger()
 
-        if not self.model.unit.is_leader():
-            self.log.info("Not a leader, skipping set_pod_spec")
-            self.model.unit.status = ActiveStatus()
-            return
-
-        self._stored.set_default(secret_key=gen_pass())
-
-        try:
-            self.interfaces = get_interfaces(self)
-        except NoVersionsListed as err:
-            self.model.unit.status = WaitingStatus(str(err))
-            return
-        except NoCompatibleVersions as err:
-            self.model.unit.status = BlockedStatus(str(err))
-            return
-        else:
-            self.model.unit.status = ActiveStatus()
+        self._stored.set_default(secret_key=_gen_pass())
 
         self.image = OCIImageResource(self, "oci-image")
 
-        self.framework.observe(self.on.install, self.set_pod_spec)
-        self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
-        self.framework.observe(self.on.config_changed, self.set_pod_spec)
+        for event in [
+            self.on.config_changed,
+            self.on.install,
+            self.on.upgrade_charm,
+            self.on["object-storage"].relation_changed,
+            self.on["object-storage"].relation_joined,
+        ]:
+            self.framework.observe(event, self.set_pod_spec)
```

Comment:
Same comment that in [mldm-operator](https://github.com/canonical/mlmd-operator/pull/9)